src/link.c

```diff
@@ -5,3 +5,10 @@ int fnm_nomatch = FNM_NOMATCH; /* Match failed. */
 int fnm_noescape = FNM_NOESCAPE; /* Disable backslash escaping. */
 int fnm_pathname = FNM_PATHNAME; /* Slash must be matched by slash. */
 int fnm_period = FNM_PERIOD;     /* Period must be matched by period. */
+
+/* A GNU extension: the pattern is matched ignoring a case */
+#ifdef FNM_CASEFOLD
+int fnm_casefold = FNM_CASEFOLD;
+#else
+int fnm_casefold = 0;
+#endif
```

Comment:
Instead of defining `fnm_casefold` unconditionally, and only making it meaningful when compiled against a fnmatch.h that supports GNU extensions (presumably glibc), let's make the definition conditional and add a rust feature flag to bring it in to `lib.rs`.

```suggestion
/* A GNU extension: the pattern is matched ignoring a case */
#ifdef FNM_CASEFOLD
int fnm_casefold = FNM_CASEFOLD;
#endif
```

That way, if someone links the library, sets `FNM_CASEFOLD` and it compiles, they know they did something.

Reply:
A feature flag cancels the whole idea. `#ifdef` in C code allows to auto-detect a presence of the feature (e.g. casefold), while a feature flag would require a developer to know and set it in advance.

Reply:
I don't want consumers of this crate to set `FNM_CASEFOLD`, and then have it do nothing, which is what would happen if linked against a non-gnu extended version of fnmatch.

The easiest way I can see to make this process transparent to consumers of fnmatch-sys is to make GNU extensions opt-in by the consuming crate with a cfg consumable feature.

Reply:
If you can think of a different way to give consumers the same degree of safety, I'm open to hear it.

Reply:
@andy128k Experimenting shows that this works: https://github.com/iwahbe/fnmatch-sys/pull/2.